arch/arm64/core/mmu.c

```diff
@@ -629,7 +630,17 @@ static int __add_map(struct arm_mmu_ptables *ptables, const char *name,
 	__ASSERT(((virt | phys | size) & (CONFIG_MMU_PAGE_SIZE - 1)) == 0,
 		 "address/size are not page aligned\n");
 	desc |= phys;
-	return set_mapping(ptables, virt, size, desc, may_overwrite);
+	ret = set_mapping(ptables, virt, size, desc, may_overwrite);
+#ifdef CONFIG_DT_HAS_XEN_XEN_ENABLED
```

Comment:
Do you really need this `ifdef`? It is better to make this variant only one option. No need to support two different approaches.

As you pointed out, set/way  does not support L3 cache, so your approach will be beneficial for all users, not only for Xen domains.

Reply:
done